src/Iteration/NondecreasingIndices.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    NondecreasingIndices
```

Comment:
nitpicking: since there is a type called `StrictlyIncreasingIndices`, shouldn't this be `NonDecreasingIndices`? or is `Nondecreasing` a word?

Reply:
Some people write "nondecreasing" [[Merriam-Webster](https://www.merriam-webster.com/dictionary/nondecreasing), [Wolfram](https://mathworld.wolfram.com/NondecreasingFunction.html)] at least. But "non-decreasing" also exists.